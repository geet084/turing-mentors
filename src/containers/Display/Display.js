import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import { Mentor } from '../../components/Mentor/Mentor';

export class Display extends Component {
  render() {
    return (
      <main className="display">
        <Switch>
          <Route exact path='/' component={Mentor} />
          <Route path='/mentor' render={() => {
            return <div>mentor</div>
          }} />
          <Route path='/mentee' render={() => {
            return <div>mentee</div>
          }} />
        </Switch>
      </main>
    )
  }
}

export default Display;