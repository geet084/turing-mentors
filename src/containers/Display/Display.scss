.display {
  display: flex;
  justify-content: center;
  padding: 30px;
}

.display div {
  color: grey;
  font-size: 2.25rem;
  margin: 0 25px;
  &:hover {
    margin-bottom: -5px;
    border-bottom: 5px solid #12f1fc;
    cursor: pointer;
  }
}