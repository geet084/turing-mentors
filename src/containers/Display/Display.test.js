import React from 'react';
import { Display } from './Display'
import { shallow } from 'enzyme'

describe('Display', () => {
  let wrapper;

  it('should match the correct snapshot', () => {
    wrapper = shallow(<Display />)
    expect(wrapper).toMatchSnapshot()
  });
});