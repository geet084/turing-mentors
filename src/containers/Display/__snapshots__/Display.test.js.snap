// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display initial should have match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    mentors={
      Array [
        Object {
          "attributes": Object {
            "availability": Object {
              "0": Array [
                true,
                false,
                false,
              ],
              "1": Array [
                true,
                false,
                false,
              ],
              "2": Array [
                true,
                false,
                false,
              ],
              "3": Array [
                true,
                false,
                false,
              ],
              "4": Array [
                true,
                false,
                false,
              ],
              "5": Array [
                false,
                true,
                false,
              ],
              "6": Array [
                false,
                true,
                false,
              ],
            },
            "background": "Things",
            "cohort": 1401,
            "contact_details": Object {
              "email": "jeff@email.com",
              "linkedin": "jeff-casimir-839123",
              "phone": "3037313117",
              "preferred_method": "1",
              "slack": "@j3",
            },
            "current_job": "Director",
            "first_name": "Jeff",
            "identities": Array [
              1,
            ],
            "last_name": "Casimir",
            "location": "Denver, CO",
            "non_tech_skills": Array [
              "1",
              "3",
              "5",
            ],
            "program": "BE",
            "tech_skills": Array [
              "1",
              "2",
              "7",
            ],
          },
          "id": "1",
        },
      ]
    }
    sendSlack={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/about"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/contact"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/contribute"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/profile"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mentors"
        />
        <Route
          path="/mentors/:id"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mentor"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mentee"
        />
        <Route
          component={[Function]}
        />
      </Switch>,
      "className": "display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/contact"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/contribute"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/profile"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/mentors"
          />,
          <Route
            path="/mentors/:id"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/mentor"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/mentee"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/about",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/contribute",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/mentors",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/mentors/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/mentor",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/mentee",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/contact"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/contribute"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/profile"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/mentors"
          />
          <Route
            path="/mentors/:id"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/mentor"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/mentee"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        "className": "display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/contact"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/contribute"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/profile"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/mentors"
            />,
            <Route
              path="/mentors/:id"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/mentor"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/mentee"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/contribute",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/profile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/mentors",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/mentors/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/mentor",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/mentee",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
